cli/cmd/physical_design/hex.hpp

```diff
@@ -0,0 +1,59 @@
+//
+// Created by Simon Hofmann on 27.04.23.
+//
+
+#ifndef FICTION_CMD_HEX_HPP
+#define FICTION_CMD_HEX_HPP
+
+#include <fiction/algorithms/physical_design/hexagonalization.hpp>
+#include <fiction/traits.hpp>
+#include <fiction/types.hpp>
+
+#include <alice/alice.hpp>
+
+namespace alice
+{
+/**
+ * Transforms a 2DDWave-clocked Cartesian layout into a hexagonal one.
+ */
+class hex_command : public command
+{
+  public:
+    /**
+     * Standard constructor. Adds descriptive information, options, and flags.
+     *
+     * @param e alice::environment that specifies stores etc.
+     */
+    explicit hex_command(const environment::ptr& e) :
+            command(e, "Transforms a 2DDWave-clocked Cartesian layout into a hexagonal one.")
+    {}
+
+  protected:
+    /**
+     * Function to transform a 2DDWave-clocked Cartesian layout into a hexagonal one.
+     */
+    void execute() override
+    {
+        auto& gls = store<fiction::gate_layout_t>();
+
+        // error case: empty gate-level layout store
+        if (gls.empty())
+        {
+            env->out() << "[w] no gate layout in store" << std::endl;
+            return;
+        }
+        const auto& lyt_ptr = gls.current();
+
+        const auto apply_hexagonalization = [](auto&& lyt_ptr) { return fiction::hexagonalization(*lyt_ptr); };
+
+        auto lyt = std::visit(apply_hexagonalization, lyt_ptr);
```

Comment:
Right now, this will crash if the layout isn't 2DDWave-clocked. Maybe you can add another condition around lines 47-51 to only execute them if the clocking scheme matches and print an error, otherwise?